src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -43,7 +43,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
             {
                 disposables.Add(engine.CreateServer(
                     address.Scheme,
-                    address.Host,
+                    // Unix sockets use a file path, not a hostname.
+                    address.Scheme == Constants.UnixScheme ? address.Path : address.Host,
```

Comment:
Is `:` a valid character in the hostname section of a URI? Would the parser allow `http://unix:/tmp/foo` as a valid URI?